testsuite/pytests/test_stdp_synapse.py

```diff
@@ -209,42 +224,49 @@ def Kpost_at_time(t, spikes, inclusive=True):
             Kpost *= exp(-(t - t_curr) / self.tau_post)
             return Kpost
 
+        eps = 1e-6
         t = 0.
+        idx_next_pre_spike = 0
+        idx_next_post_spike = 0
+        t_last_pre_spike = -1
+        t_last_post_spike = -1
         Kpre = 0.
         weight = initial_weight
 
-        t_log = []
-        w_log = []
-        Kpre_log = []
-
-        # logging
-        t_log.append(t)
-        w_log.append(weight)
-        Kpre_log.append(Kpre)
+        t_log = list()
+        w_log = dict()
+        Kpre_log = list()
+        pre_spike_times = list()
 
         post_spikes_delayed = post_spikes + self.dendritic_delay
 
         while t < self.simulation_duration:
-            idx_next_pre_spike = -1
-            if np.where((pre_spikes - t) > 0)[0].size > 0:
-                idx_next_pre_spike = np.where((pre_spikes - t) > 0)[0][0]
+            if idx_next_pre_spike >= pre_spikes.size:
+                t_next_pre_spike = -1
+            else:
                 t_next_pre_spike = pre_spikes[idx_next_pre_spike]
 
-            idx_next_post_spike = -1
-            if np.where((post_spikes_delayed - t) > 0)[0].size > 0:
-                idx_next_post_spike = np.where((post_spikes_delayed - t) > 0)[0][0]
+            if idx_next_post_spike >= post_spikes.size:
+                t_next_post_spike = -1
+            else:
                 t_next_post_spike = post_spikes_delayed[idx_next_post_spike]
 
-            if idx_next_pre_spike >= 0 and idx_next_post_spike >= 0 and t_next_post_spike < t_next_pre_spike:
-                handle_post_spike = True
+            if t_next_post_spike >= 0 and (
+                    t_next_post_spike + eps < t_next_pre_spike or t_next_pre_spike < 0):
```

Comment:
Why do we need an epsilon here now whereas we did not before? Did the test case itself hereby fundamentally change?

Reply:
I am not entirely sure anymore if that missing epsilon actually caused any issues, but it is definitely much cleaner, as comparing floats could potentially go wrong (something like 3<3.001 -> False). I know that there was a bug in the test where a missing epsilon caused a problem with some other spike times, not sure if it was this line, though.